aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,351 @@
+
+```razor
+<Dialog @rendermode="new ServerRenderMode()" />
+```
+
+> [!NOTE]
+> During .NET 8 *Release Candidate 1*, use the following values:
+>
+> Render mode | Value
+> ----------- | -----
+> Server      | `new ServerRenderMode()`
+> WebAssembly | `new WebAssemblyRenderMode()`
+> Auto        | `new AutoRenderMode()`
+>
+> The preceding syntax will be simplified in an upcoming preview release.
+
+### Prerendering
+
+<!-- UPDATE 8.0 RC2: Remove preview remark and update code -->
+
+Prerendering is enabled by default for interactive components.
+
+To disable prerendering, pass the `prerender` flag with a value of `false`.
+
+For a routable page:
+
+```razor
+@page "{ROUTE}"
+@attribute [RenderModeServer(prerender: false)]
+```
+
+In the preceding example, the `{ROUTE}` placeholder is the route template.
+
+For a non-routable, non-page component:
+
+```razor
+<Dialog @rendermode="new ServerRenderMode(prerender: false)" />
+```
+
+> [!NOTE]
+> During .NET 8 *Release Candidate 1*, use the following values:
+>
+> Render mode | Value
+> ----------- | -----
+> Server      | `new ServerRenderMode(prerender: false)`
+> WebAssembly | `new WebAssemblyRenderMode(prerender: false)`
+> Auto        | `new AutoRenderMode(prerender: false)`
+>
+> The preceding syntax will be simplified in an upcoming preview release.
+
+### Static render mode
+
+By default components use the Static render mode. The component renders to the response stream and interactivity isn't enabled.
+
+In the following example, there's no designation for the component's render mode. Therefore, the component is *statically rendered* on the server. The button isn't interactive and doesn't call the `UpdateMessage` method when selected. The value of `message` doesn't change, and the component isn't rerendered in response to UI events.
+
+`RenderMode1.razor`:
+
+```razor
+@page "/render-mode-1"
+
+<button @onclick="UpdateMessage">Click me</button> @message
+
+@code {
+    private string message = "Not clicked yet.";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+If using the preceding component locally in a Blazor Web App, place the component in the server-side project's `Components/Pages` folder. The server-side project is the solution's project with a name that doesn't end in `.Client`. When the app is running, navigate to `/render-mode-1` in the browser's address bar.
+
+> [!NOTE]
+> The anatomy of a basic Razor component is fully explained in the <xref:blazor/fundamentals/index> article. Project structure for apps created by a Blazor template are described in the <xref:blazor/project-structure> article. Detailed components coverage is found in the *Components* articles later in the documentation.
+
+### Server render mode
+
+The Server render mode renders the component interactively from the server using Blazor Server. User interactions are handled over a SignalR connection.
```

Comment:
In RC1 the circuit will stick around after it's created. In RC2 we'll need to add that the circuit will get cleaned up after the last Server component on the page has been removed with a configurable timeout: https://github.com/dotnet/aspnetcore/pull/50170

```suggestion
The Server render mode renders the component interactively from the server using Blazor Server. User interactions are handled over a real-time connection with the browser. The circuit connection will be established when the Server component is rendered.
```